editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/properties/AdvancedBeanPropertiesSection.java

```diff
@@ -0,0 +1,303 @@
+			if (jproject.exists()) {
+				return jproject;
+			}
+		}
+		return null;
+	}
+
+	private IStatus validateMethod(IJavaProject jproject, String className, String methodName) throws JavaModelException{
+		IType foundClass = jproject.findType(className);
+		boolean foundMethod = BeanConfigUtil.hasMethod(methodName, foundClass);
+		if (!foundMethod) {
+			return ValidationStatus
+					.error("Method " + methodName + " must exist in class " + className + ".");
+		}
+		return ValidationStatus.ok();
+	}
+	
+	private IStatus validateMethod(Object value) {
+		Object control = modelMap.get(CamelBean.PROP_CLASS);
+		if (control != null) {
+			String className = (String) control;
+			String methodName = (String) value;
+			IJavaProject jproject = getJavaProject();
+			if (!Strings.isEmpty(methodName) && jproject != null) {
+				try {
+					return validateMethod(jproject, className, methodName);
+				} catch (JavaModelException e) {
+					CamelEditorUIActivator.pluginLog().logError("Error validating method for class " + className,
+							e);
+				}
+			}
+		}
+		return ValidationStatus.ok();
+	}
+	
+	private AbstractTextFieldParameterPropertyUICreator createTextFieldWithClassBrowse(final Parameter p, final Composite page) {
+		AbstractTextFieldParameterPropertyUICreator txtFieldCreator = 
+				new AttributeTextFieldPropertyUICreatorWithBrowse(dbc, modelMap, eip, selectedEP, p, page, getWidgetFactory());
+		txtFieldCreator.create();
+		createClassBrowseButton(page, txtFieldCreator.getControl());
+		return txtFieldCreator;
+	}
+
+	private AbstractTextFieldParameterPropertyUICreator createTextFieldWithMethodBrowse(final Parameter p, final Composite page) {
+		AbstractTextFieldParameterPropertyUICreator txtFieldCreator = 
+				new AttributeTextFieldPropertyUICreatorWithBrowse(dbc, modelMap, eip, selectedEP, p, page, getWidgetFactory());
+		txtFieldCreator.create();
+		createMethodBrowseButton(page, txtFieldCreator.getControl());
+		return txtFieldCreator;
+	}
+
+	/**
+	 * 
+	 * @param props
+	 * @param page
+	 * @param ignorePathProperties
+	 * @param group
+	 */
+	protected void generateTabContents(List<Parameter> props, final Composite page) {
+		props.sort(new ParameterPriorityComparator());
+		for (Parameter p : props) {
+			final Parameter prop = p;
+
+			ISWTObservableValue uiObservable = null;
+			IObservableValue<Object> modelObservable;
+			IValidator validator = null;
+
+			createPropertyLabel(toolkit, page, p);
+
+			AbstractTextFieldParameterPropertyUICreator txtFieldCreator = null;
+
+			if (p.getName().contentEquals(CamelBean.PROP_CLASS)) {
+				txtFieldCreator = createTextFieldWithClassBrowse(p, page);
+				validator = new BeanClassExistsValidator();
+			} else if (p.getName().contentEquals(CamelBean.PROP_INIT_METHOD)
+					|| p.getName().contentEquals(CamelBean.PROP_DESTROY_METHOD)) {
+				txtFieldCreator = createTextFieldWithMethodBrowse(p, page);
+				validator = (Object value) -> validateMethod(value);
```

Comment:
this will lead to this::validateMethod
it is a bit tricky here, it is possible because IValidator has been declared as a Functional Interface, so you need to a provide just a single method.
i'm wondering if the best would not be to extract a specific class which implements IValidator. it will allow to create a unit test for the validation and have less code in the AdvancedBeanPropertiesSection class itself.

What do you think?